src/control/server/server.go

```diff
@@ -201,18 +201,10 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 		return err
 	}
 
-	// TODO: Move any ownership changes into the privileged binary as necessary.
-	if needsRespawn {
-		// Chown required files and respawn process under new user.
-		if err := changeFileOwnership(cfg); err != nil {
-			return errors.WithMessage(err, "changing file ownership")
-		}
-
-		log.Infof("formatting complete and file ownership changed,"+
-			"please rerun %s as user %s\n", os.Args[0], cfg.UserName)
-
-		return nil
-	}
+	harnessErr := make(chan error)
+	go func() {
+		harnessErr <- harness.Start(ctx, membership)
+	}()
 
-	return errors.Wrap(harness.Start(ctx, membership), "DAOS I/O Server exited with error")
+	return errors.Wrap(<-harnessErr, "DAOS harness exited with error")
```

Comment:
No change requested, but I'm curious how this approach changes the flow from making the call synchronously. I don't really see the difference here.

Reply:
it's not strictly necessary and there isn't much functional difference but at some point I'm anticipating wanting to handle messages/errors from the harness in main. that decoupling would facilitate a model where the harness can be interacted with by both the main server and gRPC handlers. so whilst there is no functional difference now it's a step towards that decoupling which would mean the server could still operate even if there were problems with the harness (as it doesn't add current value I will revert if you want).

Reply:
I have a different concern about this... It's generally preferable to handle the concurrency inside the concurrent method rather than just throwing a goroutine around a method that may or may not have been designed for concurrency. In other words, if the guts of harness.Start() are intended to be run in a new goroutine, then the method should be adjusted to reflect this, along with handling all of the control/error stuff. You could make harness.Start() accept an error channel to be monitored by the main goroutine, similar to the pattern that we use for starting individual instances.

That way if there is an error in setting things up prior to launching the new goroutine, Start() can exit and return an error synchronously; otherwise you can just block on the error channel after normal startup.

Reply:
@tanabarr I'm fine with either reverting for now, or taking the approach @mjmac talks about.

Reply:
will revert and address when needed